src/wymeditor/editor/dialogs.js

```diff
@@ -0,0 +1,501 @@
+"use strict";
+/**
+    editor.dialog
+    =============
+
+    Open a dialog box
+*/
+WYMeditor.editor.prototype.dialog = function (
+    dialogName,
+    pDialogWindowFeatures,
+    pBodyHtml
+) {
+    var wym = this,
+        i,
+        DIALOGS = WYMeditor.DIALOGS,
+        dialog,
+        dialogWindowFeatures,
+        wDialog,
+        htmlStrReplacements,
+        dialogHtml,
+        doc,
+        selectedContainer,
+        options = wym._options;
+
+    if (
+        DIALOGS.hasOwnProperty(dialogName) !== true &&
+        typeof pBodyHtml !== "string"
+    ) {
+        throw "No such dialog and no body HTML provided";
+    }
+
+    dialog = DIALOGS[dialogName];
+
+    // Return `false` early if this dialog should not open. Use the dialog's
+    // own function to check this.
+    if (
+        dialog &&
+        dialog.shouldOpen.call(wym) !== true
+    ) {
+        return false;
```

Comment:
Nice. I like the early exit and delegating the `shouldOpen` decision to the individual dialog object.